### PR TITLE
add delete-gauges flag to support re-sending a value when there was no u...

### PR DESCRIPTION
**Pull request by markrechler**
This is adding a flag for a config option that statsd normally supports:
https://github.com/etsy/statsd/blob/master/docs/metric_types.md#gauges

### CODE REVIEW

**Comment by markrechler** [comment]
Did not notice, this was actually done in #58. This has tests for the case where the flag is set to false as well.

**Comment by jehiah** [comment]
I guess the difference being we don't wan't "unchanged form last value" we want "any update or not" even if it's the same value.

**Comment by markrechler** [comment]
Updated.

File: statsdaemon.go [hunk 1]
```diff
@@ -269,12 +270,17 @@ func processGauges(buffer *bytes.Buffer, now int64) int64 {
 	for g, c := range gauges {
 		lastValue, ok := trackedGauges[g]
 
-		if ok && c == lastValue {
+		if ok && c == math.MaxUint64 && *deleteGauges {
 			continue
+		} else if ok && lastValue > 0 && c == math.MaxUint64 && *deleteGauges == false {
```
Comment (jehiah) [inline statsdaemon.go]: I think this is functional, but the length of these if statements is getting hard to parse. Any ideas for simplifying? Typically idiomatic Go moves longer if/else if/else if/else into a switch block to simplify.... (my attempt below)

```go
currentValue := c
lastValue, hasLastValue := trackedGauges[g]
var hasChanged bool
if c != math.MaxUint64 {
    hasChanged = true
}
switch {
    case hasChanged:
        fmt.Fprintf(buffer, "%s %d %d\n", g, currentValue, now)
        trackedGauges[g] = c
        gauges[g] = math.MaxUint64
        num++
    case !hasChanged && hasLastValue &&  !*deleteGauges:
        fmt.Fprintf(buffer, "%s %d %d\n", g, lastValue, now)
        num++
    default:
        continue
}
```

Comment (jehiah) [inline statsdaemon.go]: also i'm thinking that `s/trackedGauges/lastGaugeValue` might be a better variable name.

**Comment by markrechler** [comment]
Thanks for the tips, modified per comments.
